### PR TITLE
chore: inline build variables with custom simple macro

### LINKED ISSUE #540: Improve build-conditional logic code with a custom macro
There are a few places in downshift that we do stuff like this:

```javascript
preval`module.exports = process.env.BUILD_REACT_NATIVE !== 'true'`
```

For example:

https://github.com/paypal/downshift/blob/4b659d083f3ec0c7709b9bc9a060bab7438fdbc4/src/downshift.js#L249-L255

I'm pretty sure we could write a custom [macro](https://github.com/kentcdodds/babel-plugin-macros) to make this simpler:

```javascript
import {isReactNative} from './env.macro'

// ...

  scrollHighlightedItemIntoView() {
    /* istanbul ignore else (react-native) */
    if (!isReactNative) {
      const node = this.getItemNodeFromIndex(this.getState().highlightedIndex)
      this.props.scrollIntoView(node, this._rootNode)
    }
  }
```

I'm fairly certain all we'd need to do is replace `isReactNative` with `true` or `false` and rely on rollup's dead code elimination, or perhaps add a babel plugin for dead code elimination. Either that, or we could just eliminate the code ourselves, though I feel like that would make it harder to maintain, so I'd prefer the alternatives.

### PR DESCRIPTION
**Pull request by Andarist**
fixes #540

Some snapshot sizes are bigger, some are smaller. This PR should remove some bytes in all the cases, gonna investigate this in a moment - I suspect those differences might be caused by newer versions of downshift's deps and not downshift itself, but cant be sure right now.

EDIT:// yeah, running the build on master with fresh deps updates the snapshot, so size regressions are not caused by this PR

@TrySound would it be possible to output additional snapshot sizes for the lib itself with all deps as external? Both numbers are valuable (with externals bundled in and without them)

### CODE REVIEW

**Comment by TrySound** [comment]
@Andarist umd is a result with bundled dependencies, esm is a result without node_modules treated as externals.

**Comment by TrySound** [comment]
Lockfile could make size snapshot more stable. Yarn is the best at this. Npm sometimes works quite weird with locking dependencies.

**Comment by Andarist** [comment]
> @Andarist umd is a result with bundled dependencies (except peers usually), esm is a result without node_modules treated as externals.

Oh, makes a lot of sense that 😅 I've come to wrong conclusion when I've pinged u, thanks for clarifying this!

The true snapshot diff for this PR can be seen [here](https://github.com/paypal/downshift/compare/e3d23e3d585f19ed8181fdb4ac36a701685988fa...baa76f387b88f37970fad0a1aba40c604ba212f5#diff-29fd455fe4a1c8d19c5e5b1420ce14fa)

File: src/downshift.js [hunk 1]
```diff
@@ -248,7 +248,7 @@ class Downshift extends Component {
 
   scrollHighlightedItemIntoView() {
     /* istanbul ignore else (react-native) */
-    if (preval`module.exports = process.env.BUILD_REACT_NATIVE !== 'true'`) {
+    if (!isReactNative) {
```
Comment (kentcdodds) [inline src/downshift.js]: Out of curiosity... With this new macro, can we remove these Istanbul ignore comments?

Comment (Andarist) [reply src/downshift.js]: Maybe, i kinda dont know how those comments work **exactly**, not sure how to check if they can be removed (like i dont know what kind of outcome should i expect and where)

Comment (kentcdodds) [reply src/downshift.js]: Try to remove the comment, then run `npm run test -- --coverage` and see if we still get a 100% coverage report.

**Comment by Andarist** [comment]
Ok, so the coverage should be fixed now (I can confirm that istanbul comments cannot be removed). I've also rebased the work on top of the master so there is no conflict now.

File: src/downshift.js (hunk 1 above)
Comment (kentcdodds) [reply src/downshift.js]: Rats! Istanbul uses a babel plugin to instrument the code for coverage and I've found it's been a little tough to make it play nicely with other babel plugins at times :-(

Comment (Andarist) [reply src/downshift.js]: https://github.com/paypal/downshift/pull/542#issuecomment-412681801

it seems to work ok now, CI reports TS task failing, will have to investigate this in following days

**Comment by Andarist** [comment]
I've check what is going with Travis - the same validate script is working for me on my machine (well... actually I get some linting errors cause by eslint's `import/no-unresolved`). 

The build is failing with 137 code and based on found info in the internet the most likely cause is Out of memory error. Not sure what we should do about it right now.

**Comment by kentcdodds** [comment]
I think I've fixed it with this: https://github.com/paypal/downshift/commit/d733cf74f94e6a5a8eabb9ae820e753b8d952b4b

**Comment by kentcdodds** [comment]
There could be something we could do in a way that's built-into kcd-scripts. Maybe we could use jest-worker for the validate script: https://github.com/facebook/jest/tree/master/packages/jest-worker

**Comment by kentcdodds** [comment]
I _think_ this is good to go. Could you rebase onto master to be sure? Thanks!

**Comment by Andarist** [comment]
Rebased and green on CI, thanks! 🎉

**Pull request merged**